components/layout_2020/flow/mod.rs

```diff
@@ -69,6 +69,17 @@ impl BlockContainer {
     }
 }
 
+fn block_size_is_zero_or_auto(size: &Size, containing_block: &ContainingBlock) -> bool {
+    match size {
+        Size::Auto => true,
+        Size::LengthPercentage(ref lp) => {
+            // TODO: Should this resolve definite percentages? Blink does it, Gecko and WebKit don't.
+            lp.is_definitely_zero() ||
+                (lp.0.has_percentage() && containing_block.block_size.is_auto())
+        },
+    }
+}
```

Comment:
This is a huge nit, but maybe this free function should go at the bottom of the file instead of amidst the definition and implementation of data structures?